File: src/LongestHarmoniousSubsequence/LongestHarmoniousSubsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    var hash = new Map();
    for (var i of nums) {
      if(hash.has(i))  hash.set(i,hash.get(i)+1);
      else hash.set(i,1);
      
    }
    var res = 0;
    for (var [key,value] of hash) {
        if(hash.has(key+1)) {
           res = Math.max(res, value + hash.get(key+1)); 
        }
    }
    return res;

};

File: src/LongestHarmoniousSubsequence/LongestHarmoniousSubsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Map<Long, Integer> map = new HashMap<>();
        for(long num : nums) 
            map.put(num, map.getOrDefault(num, 0) + 1);
        
        int res = 0;
        for(long key : map.keySet()) {
            if (map.containsKey(key+1)) {
                res = Math.max(res, map.get(key) + map.get(key+1));
            }
        }
        return res;
    }
    
}